### ISSUE TITLE
Support running in the background on Linux.

### ISSUE THREAD

**Issue by AbrarSL** [transmission/transmission#5484]
### What is the issue?

See https://github.com/flatpak/xdg-desktop-portal/pull/901
GNOME 44 now has a UI for apps running in the background.
![image](https://user-images.githubusercontent.com/69783006/235598664-a68ebe4c-c69c-4ab8-96ba-4ec2f9f2ad4f.png)

### Which application of Transmission?

GTK+ app on Linux, BSD, etc.

### Which version of Transmission?

4.0.3 (6b0e49bbb2)

**Comment by killemov** [comment]
Or you could just run transmission-daemon on startup and connect with it trough web-UI or local app.

**Comment by CapeCrusader321** [comment]
@AbrarSL  @killemov    There is no need of this as already transmission offers a panel widget/applet for both gtk and qt plus for God's sake Transmission isnt Libadwaita Yet its based upon Gtk 4, the gnome background service you mentioned is helpful for libadwaita apps only as libadwaita dont support panel applets etc.
So, Transmission currently and maybe even in future wont need that.

**Comment by AbrarSL** [comment]
@CapeCrusader321 GTK4 removed tray icons support.
![Screenshot from 2023-05-02 21-06-49](https://user-images.githubusercontent.com/69783006/235715433-0f3cd489-7ec8-49e3-8649-f46915412e10.png)
Whether an app uses libadwaita or not is irrelevant as it has nothing to do with the background portal. Supporting this portal will not degrade the experience on any Linux-based platform that doesn't support the portal.

**Comment by ispanos** [comment]
![image](https://github.com/transmission/transmission/assets/42875202/3f874a92-3559-464e-8d77-eae4af841f6f)

Is this implemented already? I am using the flatpak from flathub and it can run in the background using the `-m` flag. However, once I open the window, I can't make it go back to running in the background.

**Comment by nekohayo** [comment]
Nope, as far as I can tell from this ticket and the lack of a setting for it in the GTK4  version of Transmission's GUI, this has not been implemented yet. The screenshot above is just the OS trying to "guess" the state of Transmission, but it's better for the app to explicitly control that.

Supporting this is more than a "tray icon" feature, it's a toolkit-independent OS-level feature to indicate the state (and you can do things like use the `SetStatus` function to indicate the reason why the app is running in the background, for example maybe "Downloading 4 torrents", "Seeding 6 torrents", etc.). QBittorrent needs to do the same thing too, ideally: https://github.com/qbittorrent/qBittorrent/issues/22304

The "Fragments" app decided to not allow running in the background at all, to avoid accidents: https://gitlab.gnome.org/World/Fragments/-/issues/30#note_1909051

If Transmission's desktop GUI wants to support running in the background, it would be good for it to implement the new FreeDesktop spec, for which the API is [documented here](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.Background.html). It provides a better-standardized cross-DE mechanism (instead of the unstandardized [mess](https://blog.tingping.se/2019/09/07/how-to-design-a-modern-status-icon.html) there was previously), provides a path forward to be compatible with Wayland (as per the [discussion leading to the standard's specification](https://gitlab.freedesktop.org/xdg/xdg-specs/-/issues/84)) and security sandboxing.

That spec is already implemented by a couple of desktop environments (if https://wiki.archlinux.org/title/XDG_Desktop_Portal is any indication, as of early 2025 this is at least supported by KDE, Deepin, Cinnamon, GNOME, and possibly others).

**Comment by nekohayo** [comment]
Possibly the combination of the current issue here and offering the tray icon's actions as menu items on the launcher (see #7478) could reduce the dependency on traditional system tray icons, which historically seems to cause a whole class of problems (issues #36, #3312, #5774, #279, #5680, etc.).

The backgrounding could also be done as part of the close confirmation dialog in #7085 (i.e. you ask whether the user wants to background the app or close it entirely at that time), that way you could even save yourself from implementing the extra switch in the preferences dialog, if you prefer.

**Comment by nekohayo** [comment]
To clarify one aspect of this: based on the portal's current technical limitations, the situation in GNOME is that this [only](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/3699) makes a visually noticeable difference for apps that are running within Flatpak's sandboxed environment. If it's a different kind of package, then as far as I know the current implementation is not a panacea, sadly.

**Comment by MorsMortium** [comment]
As a tip, for tray icon https://github.com/charlesrocket/libstray works well in GTK4 applications, it has a C header which contains the implementation too
